scribus/plugins/import/pdf/slaoutput.cpp

```diff
@@ -3896,3 +3926,680 @@ bool SlaOutputDev::checkClip()
+#endif
+	}
+	else if (adjunctGreater(point.y(),lastXY.y(),lineBaseXY.y()))
+	{
+		if (coLinera(point.y(), lineBaseXY.y())) //PDF never deviates from the line when it comes to colenear
+		{
+			// were back on track
+			pass = FrameworkLineTests::STYLENORMALRETURN;
+#ifdef DEBUG_TEXT_IMPORT
+			qDebug() << "STYLENORMALRETURN";
+#endif
+		}
+		else
+		{
+			//TODO: this character has overflowed the height, or is still superscript just not so much
+			pass = FrameworkLineTests::STYLESUPERSCRIPT; //could be STYLEBELOWBASELINE
+#ifdef DEBUG_TEXT_IMPORT
+			qDebug() << "STYLESUBSCRIPT point:" << point << " lastXY:" << lastXY << " lineBaseXY:" << lineBaseXY;
+#endif
+			//qDebug() << "STYLESUBSCRIPT: ";
+		}
+	}
+	else {
+		//TODO: We need to calculate things like new parargraphs and left hand justification
+		if (closeToX(point.x(), textRegioBasenOrigin.x()))
+		{
+			if (closeToY(point.y(), lastXY.y()) && !coLinera(point.y(), lastXY.y()))
+			{
+				//TODO: We need to calculate things like new parargraphs and left hand justification
+				if ((textRegionLines.size() >= 2)) //TODO: Need to setup some parameters replating to width matching, they mainly relate to justfication  && closeToX(textRegionLines[textRegionLines.size() - 2].baseOrigin.x() +  textRegionLines[textRegionLines.size() - 2].width, maxWidth))
+				{
+					//TODO: add a new line and update the deltas
+					pass = FrameworkLineTests::NEWLINE;
+#ifdef DEBUG_TEXT_IMPORT
+					qDebug() << "NEWLINE1 point:" << point << " _lastXY:" << lastXY << " origin: " << textRegioBasenOrigin <<  " this:" << this << " linespacing: " << lineSpacing;
+
+#endif // DEBUG
+				}   // we only have the first line so far, so pass without much of a test.
+				else if (textRegionLines.size() == 1) 
+				{
+					pass = FrameworkLineTests::NEWLINE;
+#ifdef DEBUG_TEXT_IMPORT
+					qDebug() << "NEWLINE2 point:" << point << " _lastXY:" << lastXY << " origin: " << textRegioBasenOrigin << " this:" << this << " linespacing: " << lineSpacing;
+#endif
+				}
+			}
+		}
+		else
+		{
+#ifdef DEBUG_TEXT_IMPORT
+			qDebug() << "NEWLINE2 oops:"<<point << ":"<<textRegioBasenOrigin << ":" << lineSpacing;
+#endif
+		}
+	}
+	return pass;
+}
+
+// Just perform some basic checks to see if newPoint can reasonably be asscribed to the current textframe.
+TextRegion::FrameworkLineTests TextRegion::isRegionConcurrent(QPointF newPoint)
+{	
+	if (glyphs.empty())
+	{
+		lineBaseXY = newPoint;
+		lastXY = newPoint;
+	}
+
+	//TODO: I need to write down which ones we want so I can work it all out, for now just some basic fuzzy matching support.
+	bool xInLimits = false;
+	if (closeToX(newPoint.x(), lastXY.x()))
+	{
+		xInLimits = true;
+	}
+	else
+	{
+		qDebug() << "X out of limits " << lastXY << " : " << newPoint;
+	}
+	bool yInLimits = false;
+	if (closeToY(newPoint.y(), lastXY.y()))
+	{
+		yInLimits = true;
+	}
+	FrameworkLineTests pass = linearTest(newPoint, xInLimits, yInLimits);

```

Comment:
```cpp
return linearTest(newPoint, closeToX(newPoint.x(), lastXY.x()), closeToY(newPoint.y(), lastXY.y())
```

(if it's too long, put it in multiple lines, as you do at other places)

or, if you plan to add more complex tests:

```cpp
bool xInLimits = closeToX(newPoint.x(), lastXY.x());
bool yInLimits = closeToY(newPoint.y(), lastXY.y());
return linearTest(newPoint, xInLimits, yInLimits);
```

but i wonder if it wouldn't be clearer to do

```cpp
return linearTest(newPoint, newPoint, lastXY);
```

and modify `linearTest()`.

since in `linearTest()` it might be a bit more clear what the "in limits" mean...

Reply:
I was going to have different versions of closeToX for moving as opposed to adding a glyph. I'm not sure but i may still do.

Reply:
moveto is always called before addglyph